client/devices/client.go

```diff
@@ -57,81 +59,109 @@ type Client struct {
 	url            string
 	devicesListURL string
 	client         *http.Client
+	output         io.Writer
 }
 
 func NewClient(url string, skipVerify bool) *Client {
 	return &Client{
 		url:            url,
 		devicesListURL: client.JoinURL(url, devicesListURL),
 		client:         client.NewHttpClient(skipVerify),
+		output:         os.Stdout,
 	}
 }
 
-func (c *Client) ListDevices(token string, detailLevel int, raw bool) error {
+func (c *Client) ListDevices(token string, detailLevel, perPage, page int, raw bool) error {
 	if detailLevel > 3 || detailLevel < 0 {
 		return fmt.Errorf("FAILURE: invalid devices detail")
 	}
 
-	body, err := client.DoGetRequest(token, c.devicesListURL, c.client)
+	req, err := http.NewRequest(http.MethodGet, c.devicesListURL, nil)
+	if err != nil {
+		return fmt.Errorf("failed to prepare request: %w", err)
+	}
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+	q := url.Values{
+		"per_page": []string{strconv.Itoa(perPage)},
+		"page":     []string{strconv.Itoa(page)},
+	}
+	req.URL.RawQuery = q.Encode()
+
+	reqDump, err := httputil.DumpRequest(req, false)
+	if err != nil {
+		return err
+	}
+	log.Verbf("sending request: \n%s", string(reqDump))
```

Comment:
this will contain the token -- we should redact that away.

Reply:
This is more or less copy-:spaghetti: of the client implementation in the common package.